### DISCUSSION TITLE
Issues related to model training

### DISCUSSION THREAD

**Discussion by dengxuezhao** [winedarksea/AutoTS#185] [Q&A]
I am a novice in time series forecasting, and I would like to consult you with some questions, thank you!
May I ask how the number of models is determined? I now have 24 months of data and want to use it to predict the data for the next three months. Do I need to change the cross-validation and the maximum number of iterations? Does the data need to be tested for stationarity in advance, or is it directly fed into AutoTS to use a randomly generated model?
In addition, I found a lot of RuntimeWarning: divide by zero encountered in divide errors during the training process. Do I need to do anything to deal with it?

**Comment by winedarksea** [comment]
No, you don't need to test for stationarity. AutoTS has preprocessing that can deal with it, if needed (only some models are sensitive to it). I do recommend you look over your historical data and make sure there aren't any weird jumps or discontinuities, or lots of missing data in recent history, as those can cause problems.

That runtime warning is probably not a concern. AutoTS tries lots of models and some of those models failing is perfectly normal - not all models work on all data. It will simply pass over the failures. However, if you run into any errors that prevent you generating forecasts, do let me know.

As for number of models, I think what you are asking about refers to `max_generations` which sets how many generations of models are tried. More generations, more models. `model_list` also affects the number of models, as it runs more the more model types you specify to try. Increasing the number of generations is always a good idea, if you have the time and compute for it.
You don't need to adjust the cross validation options but it can be a good idea. 'seasonal' is always worth trying, and more or less validations can make a difference.

**Reply by dengxuezhao** [reply]
I appreciate your suggestion, but I have made some predictions and found that after setting nonegative to true, there will still be negative numbers, and some navie model predictions seem to be less accurate, for example, lastvalue navie is not actually taking the value of the last point in time, I will analyze it later and give feedback.

**Reply by winedarksea** [reply]
LastValueNaive, _without any preprocessing_ will give the last value, but the preprocessing means it can perform far more complex modeling. Set transformer_max_depth = 0 for easier to understand but less powerful modeling.

**Reply by pauljones0** [reply]
> That runtime warning is probably not a concern. AutoTS tries lots of models and some of those models failing is perfectly normal - not all models work on all data. It will simply pass over the failures. However, if you run into any errors that prevent you generating forecasts, do let me know.

Good to know. I was getting so many when running the sample production python file I was worried that I did something wrong
